ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -310,7 +310,20 @@ + (UIImage *)loadBackButtonImageInViewController:(UIViewController *)vc withConf
         // in DEV MODE we try to load from cache (we use private API for that as it is not exposed
         // publically in headers).
         RCTImageSource *imageSource = [RNSScreenStackHeaderConfig imageSourceFromImageView:imageView];
-        RCTImageLoader *imageLoader = [subview.bridge moduleForClass:[RCTImageLoader class]];
+        RCTImageLoader *imageLoader;
+        if (subview.bridge) {
+          imageLoader = [subview.bridge moduleForClass:[RCTImageLoader class]];
+        } else {
+#ifdef RCT_NEW_ARCH_ENABLED
+          // see more info why we do this in RNSScreen dispatchEventForAnimatedObserver
+          id appDelegate = [[UIApplication sharedApplication] delegate];
+          RCTModuleRegistry *moduleRegistry =
+              (RCTModuleRegistry *)[[appDelegate valueForKey:@"_reactHost"] valueForKey:@"_moduleRegistry"];
```

Comment:
@WoLewicki, this is very interesting! 

I don't understand the full code. Why do we have to rely on the bridge/module registry for this use-case? Do you think there's any way to re-structure this so that we don't have to rely on the bridge or the module registry? 

I'm asking to know if we need to provide a better api for this. (We definitely shouldn't be doing [[appDelegate valueForKey:@"_reactHost"] valueForKey:@"_moduleRegistry"]).

Reply:
So, from what I can gather: 
- RNSScreenStackHeaderConfig is a component. 
- And it needs access to a native module: RCTImageLoader. 

As far as I'm aware, fabric components do not have access to native modules. (I think this was also the case with legacy components). And... I think there's been some hesitation about giving components access to native modules.

Let me follow up internally!

Reply:
Ok, thanks for information and waiting for follow up 🚀

Reply:
@WoLewicki, the recommended approach is to follow the footsteps of the <Image/> component.

Create a hand-written ComponentDescriptor. ComponentDescriptors have access to the contextContainer. The contextContainer contains the RCTImageLoader. 

https://github.com/facebook/react-native/blob/f7f9250f6e90a41e798a8f62738925a2859a751a/packages/react-native/ReactCommon/react/renderer/components/image/ImageComponentDescriptor.h#L20-L39

You'll also need to write a custom ShadowNode. The component descriptor will attach some c++ object to the shadow node that calls into [RCTImageLoader imageForUrl...]. And this Fabric component will be given access to the shadow node. From that shadow node, it can grab that c++ object and call the method that eventually triggers [RCTImageLoader imageForUrl...].